ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -188,9 +189,11 @@ public int persistColumnStats(Hive db, Table tbl) throws HiveException, MetaExce
     HiveTxnManager txnMgr = AcidUtils.isTransactionalTable(tbl)
         ? SessionState.get().getTxnMgr() : null;
     if (txnMgr != null) {
-      request.setValidWriteIdList(AcidUtils.getTableValidWriteIdList(conf,
-          AcidUtils.getFullTableName(tbl.getDbName(), tbl.getTableName())).toString());
       request.setWriteId(txnMgr.getAllocatedTableWriteId(tbl.getDbName(), tbl.getTableName()));
+      ValidWriteIdList writeId = AcidUtils.getTableValidWriteIdList(conf,
+          AcidUtils.getFullTableName(tbl.getDbName(), tbl.getTableName()));
+      writeId.locallyCommitWriteId(request.getWriteId());
```

Comment:
This is another change if I understand correctly. This is an optimization to try to commit the locally written WriteId.
Maybe we should handle this in a different patch. What do you think?

Reply:
None of this is optimization. It's more of a correctness thing. I mean with HA, there can be multiple instances of Cache ( CachedStore or CatalogD ) and multiple instances of HMS. Locally committing the write Id will ensure that this write Id is included in the subsequent get_* methods, which would invalidate the cache and fetch the latest from backing DB.

Reply:
Ok.. I think this is what I do not get.
What is CachedStore or CatalogD in this context. Where it is running? HS2 side? HMS side? What is the responsibility of it? Any docs, so I can learn?

Thanks,
Peter

Also, strictly speaking this is still not transactional. Either we accept that writeId generation for DDL is non-immediate, or we do it in the same transaction. Otherwise we just dig the problems deeper. Am I mistaken somewhere?